### ISSUE TITLE
running AppImage files isn't working

### ISSUE THREAD

**Issue by tycho01** [NixOS/nixpkgs#46900]
## Issue description

I tried running an [AppImage](https://appimage.org/) file using [`appimage-run`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/package-management/appimage-run/default.nix), but was met with an error. After posting this at my config repo at https://github.com/tycho01/nix-config/issues/35#issuecomment-422048579, @BenBals reported running into an identical error using a different AppImage.

### Steps to reproduce

```bash
$ appimage-run /drogon/dl/etcher-electron-1.4.4-x86_64.AppImage
/nix/store/01l9kmlwylfachsqavwln9wpzczx9pq5-appimage-exec: line 19: cd: /home/tycho/.cache/appimage-run/8ff36aec4053bad0c80186d87571dc6eac8ad0263d54befb3a5d5c0fa71297c3//squashfs-root: No such file or directory
/nix/store/01l9kmlwylfachsqavwln9wpzczx9pq5-appimage-exec: line 27: /home/tycho/.cache/appimage-run/8ff36aec4053bad0c80186d87571dc6eac8ad0263d54befb3a5d5c0fa71297c3/AppRun: No such file or directory
```

## Technical details

- [ ] Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

**Comment by xeji** [comment]
cc @tilpner who added `appimage-run` to nixpkgs (there's no listed maintainer).

**Comment by tilpner** [comment]
I suspect this is the same issue as #45973 is attempting to fix.

If you tested it on your appimage and reported back in that PR, maybe someone will feel like merging it.

**Comment by xeji** [comment]
I just merged #45973. @tycho01 please check if that solves your issue.

**Comment by tycho01** [comment]
Thanks!
